src/project/internal/projectautosaver.cpp

```diff
@@ -84,19 +82,40 @@ void ProjectAutoSaver::removeProjectUnsavedChanges(const io::path& projectPath)
 
 mu::io::path ProjectAutoSaver::projectOriginalPath(const mu::io::path& projectAutoSavePath) const
 {
+    IF_ASSERT_FAILED(io::suffix(projectAutoSavePath) == AUTOSAVE_SUFFIX) {
+        return projectAutoSavePath;
+    }
+
     return io::filename(projectAutoSavePath, false);
 }
 
 mu::io::path ProjectAutoSaver::projectAutoSavePath(const io::path& projectPath) const
 {
-    return projectPath + AUTOSAVE_SUFFIX;
+    return projectPath + "." + AUTOSAVE_SUFFIX;
```

Comment:
You can reuse appendingSuffix here

Reply:
Nice one, thanks! Fixed (other comments too).